src/WireMock.Net/Owin/AspNetCoreSelfHost.NETStandard.cs

```diff
@@ -26,21 +27,25 @@ private static void SetHttpsAndUrls(KestrelServerOptions kestrelOptions, IWireMo
                 {
                     kestrelOptions.ListenAnyIP(urlDetail.Port, listenOptions =>
                     {
-                        if (wireMockMiddlewareOptions.CustomCertificateDefined)
+                        listenOptions.UseHttps(options =>
                         {
-                            listenOptions.UseHttps(CertificateLoader.LoadCertificate(
-                                wireMockMiddlewareOptions.X509StoreName,
-                                wireMockMiddlewareOptions.X509StoreLocation,
-                                wireMockMiddlewareOptions.X509ThumbprintOrSubjectName,
-                                wireMockMiddlewareOptions.X509CertificateFilePath,
-                                wireMockMiddlewareOptions.X509CertificatePassword,
-                                urlDetail.Host)
-                            );
-                        }
-                        else
-                        {
-                            listenOptions.UseHttps();
-                        }
+                            if (wireMockMiddlewareOptions.CustomCertificateDefined)
+                            {
+                                options.ServerCertificate = CertificateLoader.LoadCertificate(
+                                    wireMockMiddlewareOptions.X509StoreName,
+                                    wireMockMiddlewareOptions.X509StoreLocation,
+                                    wireMockMiddlewareOptions.X509ThumbprintOrSubjectName,
+                                    wireMockMiddlewareOptions.X509CertificateFilePath,
+                                    wireMockMiddlewareOptions.X509CertificatePassword,
+                                    urlDetail.Host);
+                            }
+
+                            options.ClientCertificateMode = (ClientCertificateMode) wireMockMiddlewareOptions.ClientCertificateMode;
+                            if (wireMockMiddlewareOptions.SkipClientCertificateValidation)
+                            {
+                                options.ClientCertificateValidation = (_, _, _) => true;

```

Comment:
Maybe you can use `HttpClientHandler.DangerousAcceptAnyServerCertificateValidator` here?

https://learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-3.0#call-a-grpc-service-with-an-untrustedinvalid-certificate

Reply:
Unfortunately it doesn't have the same signature... DangerousAcceptAnyServerCertificateValidator takes a `HttpRequestMessage` as first arg.